sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/ProfitableTransactionSelector.java

```diff
@@ -62,6 +71,19 @@ public ProfitableTransactionSelector(
             .orElseThrow(() -> new RuntimeException("We only support a base fee market"));
   }
 
+  /**
+   * Evaluates a transaction before processing. Checks if it is profitable using its gas limit. If
+   * the transaction was found to be unprofitable during a previous block creation process, it is
+   * retried, since the gas price market could now make it profitable, but only a configurable
+   * amount of these transactions is retried each time, to avoid that they could potentially consume
+   * all the time allocated to block creation.
+   *
+   * @param evaluationContext The current selection context.
+   * @return TX_UNPROFITABLE_UPFRONT if the transaction is not profitable upfront,
+   *     TX_UNPROFITABLE_RETRY_LIMIT if the transaction was already found to be unprofitable, and
+   *     there are no more slot to retry past unprofitable transactions during this block creation
+   *     process, otherwise SELECTED.
+   */
```

Comment:
From the implementation below, I understand that unprofitableRetries is for all transactions. By default we can retry 10 times one specific or different transactions. This counter is initialized to 0 for each block creation while the cache remain static. Is my understanding correct ?

Reply:
Yes, a new instance of the selector is created on every new block creation process, so the cache is static to persist across block creations, while the retry counter is reset on every block creation, so we can retry at max 10 unprofitable tx per block creation.